src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -239,34 +239,47 @@ export class MockHttpSocket extends MockSocket {
     }
 
     if (response.body) {
-      const reader = response.body.getReader()
-
-      while (true) {
-        const { done, value } = await reader.read()
-
-        if (done) {
-          break
+      try {
+        const reader = response.body.getReader()
+
+        while (true) {
+          const { done, value } = await reader.read()
+
+          if (done) {
+            break
+          }
+
+          // Flush the headers upon the first chunk in the stream.
+          // This ensures the consumer will start receiving the response
+          // as it streams in (subsequent chunks are pushed).
+          if (httpHeaders.length > 0) {
+            flushHeaders(value)
+            continue
+          }
+
+          // Subsequent body chukns are push to the stream.
+          this.push(value)
         }
-
-        // Flush the headers upon the first chunk in the stream.
-        // This ensures the consumer will start receiving the response
-        // as it streams in (subsequent chunks are pushed).
-        if (httpHeaders.length > 0) {
-          flushHeaders(value)
-          continue
+      } catch (error) {
+        if (error instanceof Error) {
+          // Flush the headers on response stream errors.
+          // This way, the client still receives the "response" event,
+          // and the actual stream error is forwarded as the "error"
+          // event on the http.IncomingMessage instance.
+          flushHeaders()
+
+          // Forward response streaming errors as response errors.
+          /** @todo This doesn't quite do it. */
+          // this.destroy(error)
```

Comment:
I'm struggling to find a way to trigger this:

https://github.com/nodejs/node/blob/3a456c6db802b5b25594d3a9d235d4989e9f7829/lib/_http_incoming.js#L239

The `error` event on `http.IncomingMessage` is emitted when the socket is destroyed (implemented in the `._destroy()` method). For some reason, calling `this.destroy()` on the `HttpMockSocket` doesn't trigger that.